faster_whisper/transcribe.py

```diff
@@ -82,14 +113,538 @@ class TranscriptionInfo(NamedTuple):
     vad_options: VadOptions
 
 
+# The code below is copied from whisper-x (https://github.com/m-bain/whisperX)
+# and adapted for faster_whisper
+
+
+class BatchedInferencePipeline(Pipeline):
+
+    """
+    Huggingface Pipeline wrapper for WhisperModel.
+    Copyright (c) 2022, Max Bain
+    All rights reserved.
+    Modified by Mobius Labs GmbH
+    """
+
+    # TODO:
+    # - add support for timestamp mode
+    # - add support for custom inference kwargs
+
+    def __init__(
+        self,
+        model,
+        use_vad_model: bool = True,
+        options: Optional[NamedTuple] = None,
+        tokenizer=None,
+        device: Union[int, str, "torch.device"] = -1,
+        vad_device: Union[int, str, "torch.device"] = "auto",
+        framework="pt",
+        language: Optional[str] = None,
+        **kwargs,
+    ):
+        self.model = model
+        self.tokenizer = tokenizer
+        self.options = options
+        self.preset_language = language
+        self._batch_size = kwargs.pop("batch_size", None)
+        self._num_workers = 1
+        self.use_vad_model = use_vad_model
+        self.vad_onset = 0.500
+        self.vad_offset = 0.363
+        self.vad_model_url = (
+            "https://whisperx.s3.eu-west-2.amazonaws.com/model_weights/segmentation"
+            "/0b5b3216d60a2d32fc086b47ea8c67589aaeb26b7e07fcbe620d6d0b83e209ea/pytorch_model.bin"
+        )
+        (
+            self._preprocess_params,
+            self._forward_params,
+            self._postprocess_params,
+        ) = self._sanitize_parameters(**kwargs)
+        self.call_count = 0
+        self.framework = framework
+        if self.framework == "pt":
+            self.device = self.get_device(device)
+        else:
+            self.device = device
+
+        if self.use_vad_model:
+            self.vad_device = self.get_device(vad_device)
+
+            # load vad model and perform VAD preprocessing if needed
+            self.vad_model = self.load_vad_model(
+                vad_onset=self.vad_onset, vad_offset=self.vad_offset
+            )
+        self.chunk_size = 30  # VAD merging size
```

Comment:
Can chunk_size be configurable when initializing BatchedInferencePipeline?
chunk_size = 30 is usually fine for Latin language; For some Asian languages (i.e. Chinese, Japanese), smaller chunk_size (8-15) is better.

Reply:
this is also applicaple for `distil-whisper` models which have an optimal chunk size between 15-25

Reply:
Made configurable when initializing the Pipeline